components/webcompat_reporter/common/webcompat_reporter.mojom

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+module webcompat_reporter.mojom;
+
+struct ComponentInfo {
+  string name;
+  string id;
+  string version;
+};
+
+struct ReportInfo {
+  string? channel;
+  string? brave_version;
```

Comment:
nit: having `channel` and `brave_version` be optional looks strange. Shouldn't we always send this info? Or is it opt-in?

Reply:
the idea is: it could be sent from the front-end, but if no, we will try to fill it on the back-end, if possible